### PR TITLE
Created DELETE and PUT endpoints

### LINKED ISSUE #3: Add DELETE and PUT endpoints

### PR DESCRIPTION
**Pull request by alongreyber**
Created DELETE and PUT endpoints for both Questions and Answers. Fixes #3.

Updates to the API specification are not included in this pull request. There is a large refactoring to the API specification coming soon so it doesn't make sense to make those changes right now.

### CODE REVIEW

File: internal/robokache/delete.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// DeleteAnswer removes the specified answer. If fails if answer.question.owner != user.
+func DeleteAnswer(userEmail string, answer Answer) error {
+	// Confirm that such a question exists and we can add answers
+	question, err := GetQuestion(userEmail, answer.Question)
+	if err != nil {
+		return err
+	}
+	if !(question["owned"].(bool)) {
+		return errors.New("Unauthorized: You do not own this question and may not add answers")
```
Comment (patrickkwang) [inline internal/robokache/delete.go:54]:
```suggestion
		return errors.New("Unauthorized: You do not own this answer")
```

File: internal/robokache/delete.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+// DeleteQuestion deletes an existing question. It fails if question.owner != user.
+func DeleteQuestion(question Question) error {
+	// Confirm that such a question exists that we have permissions for
+	questionDB, err := GetQuestion(question.Owner, question.ID)
+	if err != nil {
+		return err
+	}
+	if !(questionDB["owned"].(bool)) {
+		return errors.New("Unauthorized: You do not own this question and may not add answers")
```
Comment (patrickkwang) [inline internal/robokache/delete.go:21]:
```suggestion
		return errors.New("Unauthorized: You do not own this question")
```

File: internal/robokache/main_test.go [hunk 3]
```diff
@@ -185,3 +219,23 @@ func TestNoSuchQuestion(t *testing.T) {
...
 	assert.Equal(t, http.StatusNotFound, w.Code)
 }
+
+func TestDeleteAnswer(t *testing.T) {
+	w := performRequest(
+		router, "DELETE", "/api/answers/0a?question_id=0", signedString, nil)
+	assert.Equal(t, http.StatusOK, w.Code)
+
+	// Get question and check that it doesn't exist
+	w = performRequest(router, "GET", "/api/answers/0", signedString, nil)
```
Comment (patrickkwang) [inline internal/robokache/main_test.go:229]:
```suggestion
	w = performRequest(router, "GET", "/api/answers/0a", signedString, nil)
```

**Review by patrickkwang — CHANGES_REQUESTED** [review]
Two big oddities:
1. DeleteQuestion does not take the userEmail like all of the other similar methods do. You put it in the Question object instead. I think that what you've done works, but line 21 there may be unreachable because GetQuestion will fail first. We should make this consistent with everything else. Passing userEmail around everywhere feels janky, so a more elegant solution might be nice.
2. Deleting answers should not require the corresponding question ID. Maybe we just call GetAnswer instead of GetQuestion in DeleteAnswer, or maybe we just have to get more clever with the SQL.

**Comment by alongreyber** [comment]
These changes have been added to #16.

**Pull request closed without merging**